atc/api/configserver/save.go

```diff
@@ -96,7 +96,7 @@ func (s *Server) SaveConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	_, created, err := team.SavePipeline(pipelineName, config, version, true)
+	_, created, err := team.SavePipeline(db.PipelineMetadata{Name: pipelineName}, config, version, true)
```

Comment:
I'm iffy on the naming here, since metadata elsewhere is used for fields that are shown to the user but ultimately have no significant bearing on the object itself. For example, on a container we store metadata for things like the step name - these values are shown in the API and used for querying in `fly intercept`, but they have no significance to the container itself. This metadata, on the other hand, is pretty crucial as it determines the ownership and lifecycle of the object.

Is this primarily to avoid adding optional arguments? If so, maybe we could add a second `SavePipeline` method like `SaveBuildPipeline` or something? Or maybe we could add a `build.SavePipeline` method, which would get the team/job from its existing relationships? 🤔 Just throwing ideas out there.

Reply:
+1 to having a separate method - `build.SavePipeline` sounds cool